cloud/openstack/os_keypair.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+# Copyright (c) 2013, Benno Joy <benno@ansible.com>
+# Copyright (c) 2013, John Dewey <john@dewey.ws>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+DOCUMENTATION = '''
+---
+module: os_keypair
+short_description: Add/Delete a keypair from OpenStack
+extends_documentation_fragment: openstack
+version_added: "2.0"
+description:
+   - Add or Remove key pair from OpenStack
+options:
+   name:
+     description:
+        - Name that has to be given to the key pair
+     required: true
+     default: None
+   public_key:
+     description:
+        - The public key that would be uploaded to nova and injected into VMs
+          upon creation.
+     required: false
+     default: None
+   public_key_file:
+     description:
+        - Path to local file containing ssh public key. Mutually exclusive
+          with public_key.
+    required: false
+    default: None
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Creates a key pair with the running users public key
+- os_keypair:
+      cloud: mordred
+      state: present
+      name: ansible_key
+      public_key_file: /home/me/.ssh/id_rsa.pub
+
+# Creates a new key pair and the private key returned after the run.
+- os_keypair:
+      cloud: rax-dfw
+      state: present
+      name: ansible_key
+'''
```

Comment:
That helps, but I meant in the actual doc structure: https://github.com/ansible/ansible-modules-core/blob/devel/files/stat.py#L72